tests/subprocesstest.py

```diff
@@ -94,19 +97,22 @@ def _full_method_name(self):
             modpath = self.__class__.__module__
             clsname = self.__class__.__name__
         testname = test.__name__
-        testcase_name = "{}.{}.{}".format(modpath, clsname, testname)
+        format_string = "{}.py::{}::{}" if use_pytest else "{}.{}.{}"
+        modpath = modpath.replace(".", "/" if use_pytest else ".")
+        testcase_name = format_string.format(modpath, clsname, testname)
         return testcase_name
 
     def _run_test_in_subprocess(self, result):
-        full_testcase_name = self._full_method_name()
-
         # Copy the environment and include the special subprocess environment
         # variable for the subprocess to detect.
         env_overrides = self._get_env_overrides()
+        use_pytest = self._get_use_pytest()
+        full_testcase_name = self._full_method_name(use_pytest=use_pytest)
         sp_test_env = os.environ.copy()
         sp_test_env.update(env_overrides)
         sp_test_env[SUBPROC_ENV_VAR] = "True"
-        sp_test_cmd = ["python", "-m", "unittest", full_testcase_name]
+        test_framework = "pytest" if use_pytest else "unittest"
+        sp_test_cmd = ["python", "-m", test_framework, full_testcase_name]
```

Comment:
I needed this change to be able to use fixtures in the subprocess test.

I tried running all subprocess tests with pytests, but several (~5) contrib test suites failed.